metadata-jobs/mae-consumer/src/test/java/com/linkedin/metadata/kafka/hook/GraphIndexUtilsTest.java

```diff
@@ -0,0 +1,117 @@
+import com.linkedin.common.urn.DatasetUrn;
+import com.linkedin.common.urn.Urn;
+import com.linkedin.common.urn.UrnUtils;
+import com.linkedin.data.template.RecordTemplate;
+import com.linkedin.dataset.DatasetLineageType;
+import com.linkedin.dataset.Upstream;
+import com.linkedin.dataset.UpstreamArray;
+import com.linkedin.dataset.UpstreamLineage;
+import com.linkedin.events.metadata.ChangeType;
+import com.linkedin.metadata.Constants;
+import com.linkedin.metadata.graph.Edge;
+import com.linkedin.metadata.models.AspectSpec;
+import com.linkedin.metadata.models.EntitySpec;
+import com.linkedin.metadata.models.RelationshipFieldSpec;
+import com.linkedin.metadata.models.extractor.FieldExtractor;
+import com.linkedin.metadata.models.registry.ConfigEntityRegistry;
+import com.linkedin.metadata.models.registry.EntityRegistry;
+import com.linkedin.metadata.utils.GenericRecordUtils;
+import com.linkedin.mxe.MetadataChangeLog;
+import com.linkedin.mxe.SystemMetadata;
+import org.testng.Assert;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+public class GraphIndexUtilsTest {
+
+  private static final long CREATED_EVENT_TIME = 123L;
+  private static final long UPDATED_EVENT_TIME = 234L;
+  private Urn _datasetUrn;
+  private DatasetUrn _upstreamDataset1;
+  private DatasetUrn _upstreamDataset2;
+  private static final String CREATED_ACTOR_URN = "urn:li:corpuser:creating";
+  private static final String UPDATED_ACTOR_URN = "urn:li:corpuser:updating";
+  private EntityRegistry _mockRegistry;
+  private Urn _createdActorUrn;
+  private Urn _updatedActorUrn;
+
+  @BeforeMethod
+  public void setupTest() {
+    _createdActorUrn = UrnUtils.getUrn(CREATED_ACTOR_URN);
+    _updatedActorUrn = UrnUtils.getUrn(UPDATED_ACTOR_URN);
+    _datasetUrn = UrnUtils.getUrn("urn:li:dataset:(urn:li:dataPlatform:bigquery,my-proj.jaffle_shop.customers,PROD)");
+    _upstreamDataset1 = UrnUtils.toDatasetUrn("snowflake", "test", "DEV");
+    _upstreamDataset2 = UrnUtils.toDatasetUrn("snowflake", "test2", "DEV");
+    _mockRegistry = new ConfigEntityRegistry(
+        UpdateIndicesHookTest.class.getClassLoader().getResourceAsStream("test-entity-registry.yml"));
+  }
+
+  @Test
+  public void testExtractGraphEdgesDefault() {
+    UpstreamLineage upstreamLineage = createUpstreamLineage();
+    MetadataChangeLog event = createMCL(upstreamLineage);
+
+    EntitySpec entitySpec = _mockRegistry.getEntitySpec(event.getEntityType());
+    AspectSpec aspectSpec = entitySpec.getAspectSpec(event.getAspectName());
+
+    Map<RelationshipFieldSpec, List<Object>> extractedFields =
+        FieldExtractor.extractFields(upstreamLineage, aspectSpec.getRelationshipFieldSpecs());
+
+    for (Map.Entry<RelationshipFieldSpec, List<Object>> entry : extractedFields.entrySet()) {
+      if (entry.getKey().getPath().toString().equals("/upstreams/*/dataset")) {
```

Comment:
where is this path coming from? is this a constant somewhere?

Reply:
yeah it's just the path I deduced from this specific UpstreamLineage aspect - not a constant anywhere (however I can make it a constant in this test file for readability)

Reply:
Sg!